webpack/assets/javascripts/react_app/common/helpers.js

```diff
@@ -1,11 +1,28 @@
+import debounce from 'lodash/debounce';
+
+export const debounceMethods = (context, time, methods) => {
+  methods.forEach((method) => {
+    const methodName = method.name || method;
+    const methodTime = method.time || time;
+    // eslint-disable-next-line no-param-reassign
+    context[methodName] = debounce(context[methodName], methodTime);
+  });
+};
+
+export const bindMethods = (context, methods) => {
+  methods.forEach((method) => {
+    // eslint-disable-next-line no-param-reassign
+    context[method] = context[method].bind(context);
+  });
+};
+
 export const noop = Function.prototype; // empty function
 
+export const clearSpaces = string => string.trim().replace(/\s\s+/, ' ');
```

Comment:
do we need both trip and replace?

Reply:
The trim method only removes whitespace from both ends of a string, but doesn't replace double spacing inside it.